File: src/reverse-integer/reverse-integer.py
# // https://leetcode.com/problems/reverse-integer/

class Solution:
    def reverse(self, x: int) -> int:
        negative = x < 0
        x_abs = abs(x)
        result = 0

        while x_abs:
            last_digit = x_abs % 10
            x_abs //= 10
            result = result * 10 + last_digit

        if result > 2 ** 31:
            result = 0

        return -result if negative else result
        

File: src/reverse-integer/reverse-integer.ts
// https://leetcode.com/problems/reverse-integer/

function reverse(x: number): number {
    // use a string hack, no need for modulu math here
    const str = '' + x;
    let result = 0;
    if (str.startsWith('-')) {
      result = Number(str.substring(1).split('').reverse().join('')) * -1;
    } else {
      result = Number(str.split('').reverse().join(''))
    }
      return (result > 2 ** 31 || result < -1 * 2 ** 31) ? 0 : result;
  };